### PR TITLE
Extract isosurface

### PR DESCRIPTION
**Pull request by moritzdannhauer**
Ok, I converted the module + Marching Cube.
I dont know how to create/test some of the exotic geometrical shapes though, like prisms etc.

There is some stuff not implemented yet:
1) Multi-threading
2) Geom* (e.g. GeomFastTriangles), AppendMatrix

### CODE REVIEW

**Comment by dcwhite** [comment]
Thanks, I will take a look some time this week.

File: src/Core/Algorithms/Legacy/Fields/MarchingCubes/MarchingCubes.h [hunk 1]
```diff
@@ -29,57 +29,65 @@
...
-using namespace SCIRun;
+#include <Core/Algorithms/Base/AlgorithmBase.h>
+#include <Core/Algorithms/Field/share.h>
 
-class SCISHARE MarchingCubesAlgo : public AlgoBase {
+namespace SCIRun {
+ namespace Core {
+  namespace Algorithms {
+  
+   using namespace SCIRun::Core::Datatypes;
```
Comment (dcwhite) [inline src/Core/Algorithms/Legacy/Fields/MarchingCubes/MarchingCubes.h]: Remove: no usings in headers.

File: src/Core/Algorithms/Legacy/Fields/MarchingCubes/MarchingCubes.h [hunk 2]
```diff
@@ -29,57 +29,65 @@
...
-#include <Core/Algorithms/Fields/share.h>
-
-namespace SCIRunAlgo {
+#ifdef SCIRUN4_CODE_TO_BE_ENABLED_LATER
+ #include <Core/Algorithms/Util/AlgoBase.h>
+#endif
 
-using namespace SCIRun;
+#include <Core/Algorithms/Base/AlgorithmBase.h>
+#include <Core/Algorithms/Field/share.h>
```
Comment (dcwhite) [inline src/Core/Algorithms/Legacy/Fields/MarchingCubes/MarchingCubes.h]: Wrong share.h for this file (Windows build fails due to this): use Core/Algorithms/Legacy/Fields/share.h instead

**Comment by dcwhite** [comment]
@moritzdannhauer Please add a few test networks to verify the module.

**Comment by aylakhan** [comment]
Don't worry too much about the more exotic mesh types - we still need to come up with test geometries for those.

**Comment by moritzdannhauer** [comment]
How about now?

**Comment by dcwhite** [comment]
Played with the sample net a bit and it works great. I'll do a more thorough code review next month.

**Pull request merged**